src/include/miopen/solver/implicitgemm_ck_util.hpp

```diff
@@ -114,15 +134,36 @@ ConvSolution InitInvokerFactoryNHWC(const ExecutionContext&,
     }
 
     ConvSolution result;
-    result.invoker_factory =
-        [ck_args     = CKArgsType{problem},
-         sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](const std::vector<Kernel>&) mutable {
-            return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
-                       const Handle& handle, const AnyInvokeParams& primitive_parameters) {
-                const auto& data_ctx = primitive_parameters.CastTo<CastType>();
-                auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
-                auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
+    result.invoker_factory = [ck_args     = CKArgsType{problem},
+                              sh_conv_ptr = std::shared_ptr{std::move(*ptr_iter)}](
+                                 const std::vector<Kernel>&) mutable {
+        return [ck_args = std::move(ck_args), sh_conv_ptr = std::move(sh_conv_ptr)](
+                   const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            const auto& data_ctx = primitive_parameters.CastTo<CastType>();
+            auto argument_ptr    = ck_args.MakeArgPtr(sh_conv_ptr, data_ctx.tensors);
+            auto invoker_ptr     = sh_conv_ptr->MakeInvokerPointer();
 
+            if constexpr(std::is_same<CastType, miopen::conv::WrWInvokeParams>::value)
+            {
+                auto zero           = 0.0f;
+                const auto& tensors = data_ctx.tensors;
+#if MIOPEN_BACKEND_HIP
+                HipEventPtr start = nullptr;
+                HipEventPtr stop  = nullptr;
+                if(handle.IsProfilingEnabled())
+                {
+                    ProfilingRecordStart(handle, start, stop);
+                }
+#endif
+                SetTensor(handle, tensors.dwDesc, tensors.dw, &zero);
+                invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), false});
+#if MIOPEN_BACKEND_HIP
+                if(handle.IsProfilingEnabled())
+                    ProfilingRecordStop(handle, start, stop);
+#endif
```

Comment:
Please use approach 1 from https://github.com/ROCm/MIOpen/pull/2429#discussion_r1360795493 to accumulate time.

Reply:
> Please use approach 1 from [#2429 (comment)](https://github.com/ROCm/MIOpen/pull/2429#discussion_r1360795493) to accumulate time.

Unfortunately, this approach does not work. I explained [here](https://github.com/ROCm/MIOpen/pull/2770#discussion_r1506089187)

Reply:
[resolved]